cmake/external/onnxruntime_external_deps.cmake

```diff
@@ -104,7 +104,7 @@ FetchContent_Declare(
     URL ${DEP_URL_flatbuffers}
     URL_HASH SHA1=${DEP_SHA1_flatbuffers}
     PATCH_COMMAND ${ONNXRUNTIME_FLATBUFFERS_PATCH_COMMAND}
-    FIND_PACKAGE_ARGS 1.12.0...<2.0.0 NAMES Flatbuffers
+    FIND_PACKAGE_ARGS 23.5.26...<24.0.0 NAMES Flatbuffers
```

Comment:
flatbuffers now uses a date-based version number
https://github.com/google/flatbuffers/wiki/Versioning

given that, I'm not sure that this upper bound makes sense.
if the requirement is 64-bit offset support, perhaps we can just say that 23.5.9 is the minimum version required. not sure how conservative we want to be about the max version though. what do you think, @snnn?

Reply:
Sounds good. We may choose the simpler way until someone reports problems to us.

Reply:
there are two simpler ways I can think of :)
1. no upper bound: 23.5.9+
2. version range: 23.5.9 - 23.5.26
any preference?

Reply:
no upper bound